src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Controller/ControllerGeneratorBase.cs

```diff
@@ -83,13 +84,31 @@ protected virtual IEnumerable<string> TemplateFolders
             }
         }
 
-        protected string GetControllerNamespace()
+        protected string GetDefaultControllerNamespace()
         {
             // Review: MVC scaffolding used ActiveProject's MSBuild RootNamespace property
             // That's not possible in command line scaffolding - the closest we can get is
             // the name of assembly??
             var appName = ApplicationInfo.ApplicationName;
-            return appName + "." + Constants.ControllersFolderName;
+
+            var defaultControllerNamespace = appName + "." + Constants.ControllersFolderName;
+            
```

Comment:
Issue 155 also brought a valid point : you know location of project.json don't you? xproj if it exist should be located in the same folder. If it is the only xproj in the folder you could get its <AssemblyName>, if there are several xproj's in the folder , try to find the one with name ="containing folder name" (?). In all other cases use your default existing logic..